CHANGELOG.md

```diff
@@ -1,80 +1,143 @@
+## Fleet 4.59.0 (Nov 1, 2024)
+
+### Endpoint Operations
+- Updated OpenTelemetry libraries to latest versions. This includes the following changes when OpenTelemetry is enabled:
+  - MySQL spans outside of HTTPS transactions are now logged.
+  - Renamed MySQL spans to include the query, for easier tracking/debugging.
+
+### Device Management (MDM)
+- Added UI for adding a setup experience script.
+- Added UI for the install software setup experience.
+- Added software experience software title selection API.
+- Added database migrations to support Setup Experience.
+- Added support to `fleetctl gitops` to specify a setup experience script to run and software to install, for a team or no team.
+- Added an Orbit endpoint (`POST /orbit/setup_experience/status`) for checking the status of a macOS host's setup experience steps.
+- Added service to track install status.
+- Added ability to connect a SCEP NDES proxy.
+- Added SCEP proxy for Windows NDES (Network Device Enrollment Service) AD CS server, which allows devices to request certificates.
+- Added error message on the My Device page when MDM is off for the host.
+- Added a config field to the UI for custom MDM URLs.
+- Added integration to queue setup experience software installation on automatic enrollment.
+- Added a validation to prevent removing a software package or a VPP app from a team if that software is selected to be installed during the setup experience.
+- Updated to remove a pending MDM device if it was deleted from current ABM.
+- Updated to ensure details for a software installation run are available and accurate even after the corresponding installer has been edited or deleted.
+  - **NOTE:** The database migration included with this update backfills installer data into installation details based on the currently uploaded installer. If you want to backfill data from activities (which will be more comprehensive and accurate than the migration default, but may take awhile as the entire activities table will be scanned), run this database query _after_ running database migrations:
+```sql
+UPDATE host_software_installs i
+JOIN activities a ON a.activity_type = 'installed_software'
+	AND i.execution_id = a.details->>"$.install_uuid"
+SET i.software_title_name = COALESCE(a.details->>"$.software_title", i.software_title_name),
+	i.installer_filename = COALESCE(a.details->>"$.software_package", i.installer_filename),
+	i.updated_at = i.updated_at
```

Comment:
@georgekarrv @PezHub Since we're including this SQL in our release notes please make sure it is QA'd and works as expected. Thanks!

Reply:
@georgekarrv @lukeheath I confirmed that `fleet-maintained apps` and `custom packages` are listed when running the queries but I'm not seeing VPP apps. Is that expected? (I can ask in slack as well)

`SELECT * FROM host_software_installs`
`SELECT * FROM activities WHERE activity_type = 'installed_software'`

Reply:
looks like VPP apps are stored in `vpp_apps` so makes sense they wouldn't show up